mne/encoding/feature.py

```diff
@@ -0,0 +1,250 @@
+"""A collection of classes for transforming, preprocessing, and fitting."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Transformer for creating delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    time_window : array, shape (2,)
+        Specifies a minimum/maximum delay in seconds. In this case, all
+        delays between `min(time_window)` and `max(time_window)`
+        will be calculated using `sfreq`.
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
```

Comment:
My suggestion would still (at least after our last meeting) be:

- here, do what @choldgraf thinks should be in a minimally simple encoder (e.g. "delays" workflow for lags)
- later, we try to patch it together with rERP-y stuff I would like to see (e.g. time win workflow for lags)